### PR TITLE
Give VaList its own home

### PR DESCRIPTION
**Pull request by workingjubilee**
Just rearranging things internally and reexporting.

### CODE REVIEW

File: library/core/src/ffi/va_list.rs [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+        not(target_arch = "x86_64")
+    ),
+    all(target_arch = "aarch64", target_vendor = "apple"),
+    target_family = "wasm",
+    target_os = "uefi",
+    windows,
+))]
+#[cfg_attr(not(doc), repr(transparent))] // work around https://github.com/rust-lang/rust/issues/90435
+#[lang = "va_list"]
+pub struct VaListImpl<'f> {
```
Comment (Urgau) [inline library/core/src/ffi/va_list.rs:28]: Shouldn't those platform-dependant `VaListImpl` struct be moved into the corresponding PAL?

Comment (workingjubilee) [reply library/core/src/ffi/va_list.rs:28]: nonesuch for core, currently.

**Comment by workingjubilee** [comment]
This is now just internal-only.

r? libs

**Comment by Nilstrieb** [comment]
may it live peacefully in its new home.
bors r plus

**Pull request merged**